homeassistant/config.py

```diff
@@ -1432,22 +1432,24 @@ async def async_process_component_config(  # noqa: C901
 
     # Check if the integration has a custom config validator
     config_validator = None
-    try:
-        config_validator = await integration.async_get_platform("config")
-    except ImportError as err:
-        # Filter out import error of the config platform.
-        # If the config platform contains bad imports, make sure
-        # that still fails.
-        if err.name != f"{integration.pkg_path}.config":
-            exc_info = ConfigExceptionInfo(
-                err,
-                ConfigErrorTranslationKey.CONFIG_PLATFORM_IMPORT_ERR,
-                domain,
-                config,
-                integration_docs,
-            )
-            config_exceptions.append(exc_info)
-            return IntegrationConfigInfo(None, config_exceptions)
+    if integration.platform_exists("config") is not False:
```

Comment:
This call is blocking.

Reply:
Yes it is, but its far cheaper than what importlib will do if its called so its worth the trade off. There isn't a currently isn't a good solution that doesn't flood the executor with call to do the two `stat`s.

Reply:
Ideally we come up with a way to pre do the `stat()` calls in a single executor job, but we don't know the location of the component on disk until we load it and can check `__file__` so its not simple to solve

Reply:
Please always add a code comment when there's an exception to our normal procedures, and explain why we do it that way. In this case the normal procedure would be to schedule the blocking call on the executor.

Reply:
I think I can make this particular one go away in two more turns after https://github.com/home-assistant/core/pull/112096

If I can't get rid of it soon, I'll add comments

The one in integrtion_platform is harder to solve so I'll add that one now

Reply:
https://github.com/home-assistant/core/pull/112103

Reply:
> I think I can make this particular one go away in two more turns after #112096

https://github.com/home-assistant/core/pull/112104 will add tests + more comments after sleep as its past midnight here

Reply:
https://github.com/home-assistant/core/pull/112168 will eliminate the I/O in the event loop when merged along with https://github.com/home-assistant/core/pull/112127